mrmustard/lab_dev/states/dm.py

```diff
@@ -209,32 +198,25 @@ def auto_shape(
             max_shape: The maximum shape to compute (default in ``settings.AUTOSHAPE_MAX``).
             respect_manual_shape: Whether to respect the non-None values in ``manual_shape``.
         """
-        # experimental:
-        if self.ansatz.batch_size <= 1:
-            try:  # fock
-                shape = self.ansatz.core_shape
-            except AttributeError:  # bargmann
-                if self.ansatz.num_derived_vars == 0:
-                    ansatz = self.ansatz
-                    A, b, c = (
-                        (ansatz.A[0], ansatz.b[0], ansatz.c[0])
-                        if ansatz.batch_shape != ()  # tensorflow
-                        else ansatz.triple
-                    )
-                    ansatz = ansatz / self.probability
-                    shape = autoshape_numba(
-                        math.asnumpy(A),
-                        math.asnumpy(b),
-                        math.asnumpy(c),
-                        max_prob or settings.AUTOSHAPE_PROBABILITY,
-                        max_shape or settings.AUTOSHAPE_MAX,
-                    )
-                    shape = tuple(shape) + tuple(shape)
-                else:
-                    shape = [settings.AUTOSHAPE_MAX] * 2 * len(self.modes)
-        else:
-            warnings.warn("auto_shape only looks at the shape of the first element of the batch.")
-            shape = [settings.AUTOSHAPE_MAX] * 2 * len(self.modes)
+        if self.ansatz.batch_shape:
+            raise NotImplementedError("Batched auto_shape is not implemented.")
```

Comment:
Does it make sense to compute the batch of all the states and then keep the largest on each mode? Perhaps we can put a limit to how many states in the batch we are willing to compute the shape for?